src/uu/join/src/join.rs

```diff
@@ -591,11 +591,11 @@ impl<'a> State<'a> {
     }
 }
 
-#[uucore::main]
-#[allow(clippy::cognitive_complexity)]
-pub fn uumain(args: impl uucore::Args) -> UResult<()> {
-    let matches = uu_app().try_get_matches_from(args)?;
-
+/// Parses the command-line arguments and constructs a `Settings` struct.
+///
+/// This function takes the matches from the command-line arguments, processes them,
+/// and returns a `Settings` struct that encapsulates the configuration for the program.
+fn parse_settings(matches: &clap::ArgMatches) -> UResult<Settings> {
```

Comment:
Unfortunately, the refactorings are not enough and the `cognitive_complexity` lint still shows a warning.
```suggestion
#[allow(clippy::cognitive_complexity)]
fn parse_settings(matches: &clap::ArgMatches) -> UResult<Settings> {
```

Reply:
oh, maybe we should break the clippy task on this too

Reply:
Hm, what do you mean with "break the clippy task"? Currently, the lint still shows a warning.

Reply:
like this:
https://github.com/uutils/coreutils/pull/5379

Reply:
Sorry, I still don't understand what you mean :| #5379 has been merged in the meantime and it shows an error for this PR (for some reason the job doesn't fail):
```
error: the function has a cognitive complexity of (12/10)
   --> src/uu/join/src/join.rs:622:4
    |
622 | fn parse_settings(matches: &clap::ArgMatches) -> UResult<Settings> {
```